### PR TITLE
[DG] ADY-38 added controller and start adyen configuration validation

### PR DESCRIPTION
**Pull request by pandemicGander**

### CODE REVIEW

File: src/main/java/com/adyen/mirakl/startup/AdyenStartupValidator.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+            .collect(Collectors.toMap(NotificationConfigurationDetails::getNotifyURL, Function.identity()));
+
+        //filter for notification ids we care about
+        final GetNotificationConfigurationListResponse notificationConfigurationList = adyenNotification.getNotificationConfigurationList();
+        final Map<String, Long> notifyUrlsToIds = notificationConfigurationList.getConfigurations().stream()
+            .filter(configurationDetail -> currentNotificationSetup.containsKey(configurationDetail.getNotifyURL()))
+            .collect(Collectors.toMap(NotificationConfigurationDetails::getNotifyURL, NotificationConfigurationDetails::getNotificationId));
+
+        //transfer the notification ID
+        notifyUrlsToIds.forEach((desc, id) -> currentNotificationSetup.get(desc).setNotificationId(id));
```
Comment (lancergr) [inline src/main/java/com/adyen/mirakl/startup/AdyenStartupValidator.java]: minor: we should have (url, id) here instead of (desc, id), right?

Comment (pandemicGander) [reply src/main/java/com/adyen/mirakl/startup/AdyenStartupValidator.java]: yep
will update

**Comment by rikterbeek** [comment]
Think we need extra security measures in place:
"notifyUsername":"testUserName",
"notifyPassword":"testPassword",
"sslProtocol":"TLS2.1"

**Comment by pandemicGander** [comment]
@rikterbeek I'll update

**Comment by rikterbeek** [comment]
Great no further remarks or blockers on my side

**Pull request merged**